packages/http/src/decorators.ts

```diff
@@ -188,7 +188,7 @@ export const $path: PathDecorator = (
     type: "path",
     explode: userOptions.explode ?? false,
     allowReserved: userOptions.allowReserved ?? false,
-    style: userOptions.style ?? "simple",
+    style: userOptions.style ?? (entity.optional ? "path" : "simple"),
```

Comment:
I think this resolution has to be done later, just found out this currently create a bug so lets try not to make it worse

https://github.com/microsoft/typespec/issues/6571